cpp/include/cudf/io/text/byte_range_info.hpp

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright (c) 2021, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include <cudf/detail/utilities/integer_utils.hpp>
+
+#include <limits>
+#include <vector>
+
+namespace cudf {
+namespace io {
+namespace text {
+
+struct byte_range_info {
```

Comment:
This could use some doxygen. Some text like what you put in the PR description.

Reply:
Still would like to capture the PR description and examples somewhere in the source code.